javascript/atoms/dom.js

```diff
@@ -601,7 +603,7 @@ bot.dom.isShown = function(elem, opt_ignoreOpacity) {
         return false;
       }
       var parent = bot.dom.getParentElement(e);
-      return !parent || displayed(parent);
+      return parent && displayed(parent);
```

Comment:
does this reversal seem right? it seems before we were assuming a parentless element was visible. we should be assuming the opposite

Reply:
an element without a parentNode could be a disconnected node.

also, as far as you go up the chain, there will be a parentNode, until it is `#document`. 

`body` has a parentNode of `document`, and we check that elsewhere (we return true somewhere else when `node === document`). so i dont see a situation where we would end up with a parentless node that is still somehow part of the document.

Reply:
we're walking up parent nodes. this calls `getParentNodeInComposedDom` which (as seen below) uses `parentNode`

Reply:
The line immediately preceding this one:

`var parent = bot.dom.getParentElement(e);`

[getParentElement](https://github.com/SeleniumHQ/selenium/blob/master/javascript/atoms/dom.js#L363) returns the parent element or null.

Reply:
ah my mistake, it shouldn't really be doing either to be honest. an element without a parent is more likely to be a disconnected element than document, its not the best assertion really.

we are better off asserting that it is `document` or not. if it isn't we do this line as is, otherwise we return true.

Reply:
in actual fact @jleyba  why do we have this branch at all? `getParentNodeInComposedDom` works without shadow DOM support (it checks for support internally). why not drop this `else` completely and use the same function?